### ISSUE TITLE
Unused third party dependencies

### ISSUE THREAD

**Issue by gbehrmann** [opensciencegrid/privilege-xacml#11]
Hi,

I got some questions about the dependencies specified in the main pom. Are the following actually used:

joda-time
velocity
xerces
xmlParserAPIs
xalan
wsdl4j

wsdl4j is used during compile time for code generation, but is it actually a runtime dependency?

Some of the dependencies are imported by opensaml too, but if there is no direct dependency in privilege-xacml then they should not be specified, imo.

Hope you can help clear this up.

Thanks,

/gerd

**Comment by bbockelm** [comment]
Hi Gerd,

When I run `mvn dependency:analyze`, it picks up all those as dependencies as a used during compile time.  For example, focusing on `joda-time`, I get a complaint if I drop it from the direct dependencies.  Really, I think this indicates I'm compensating for a missing indirect dependency.  Do you have any tips on how to nail down the problematic POM?

Certainly, I think `wsdl4j` can be moved to compile-only.  

Brian

**Comment by gbehrmann** [comment]
Regarding `joda-time`, is that a complaint at runtime? If I comment out that part in `pom.xml`, it still compiles fine.

I should add that I don't have a good way to test the lib myself - my testing procedure is to ask our friends at FNAL to run tests :-)

**Comment by bbockelm** [comment]
The complaint from dep-analyze is at compile time.  I suspect that one of our dependency JARs has code which uses it - but in a package not actually used - which the static analyzer is picking up.

I guess I can put some exceptions in there to suppress the warnings - but it does make me a bit nervous since I don't know how they are used in the dependency packages.

I also only test through a package (GUMS) that builds on top of privilege-xacml.  Unfortunately, the original/prior developer's only tests require access to their long-deleted home directory.  Part of the fun...

**Comment by gbehrmann** [comment]
> Unfortunately, the original/prior developer's only tests require access to their long-deleted home directory. Part of the fun...

Yeah, that's surprisingly common - JGlobus tests had similar expectations.
